### PR TITLE
feat: Thanos and Minio with updated Velero config

### PR DESCRIPTION
**Pull request by srodenhuis**
This PR brings:

- Thanos and Minio in an out-of-the-box remote write setup
- Configure Prometheus for remote write
- Install Minio with provisioning (creating buckets and policies for all Otomi apps to consume object storage
- Splitting up cloud and storage provider for Velero to support using in-cluster Minio on other clouds

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [ x ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [ x ] Helm releases are meeting otomi's baseline security policies, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

**Comment by staticvoid255** [comment]
Hi Sander - I noticed that if you try to install Minio and Thanos at the same time, that Core will try to install Thanos first. This then causes the Drone pipeline to hang as Thanos is stuck in error trying to reach a Minio instance that doesn't yet exist.
Screenshot 2022-11-10 at 15 13 16

**Comment by Morriz** [comment]
> Hi Sander - I noticed that if you try to install Minio and Thanos at the same time, that Core will try to install Thanos first. This then causes the Drone pipeline to hang as Thanos is stuck in error trying to reach a Minio instance that doesn't yet exist.
> Screenshot 2022-11-10 at 15 13 16

Any app that has a dependency on another service should wait for that service in a pre-install job. Easiest way to do that is to add a `needs:` dependency to the helmfile release by deploying another `wait-for` chart. Both releases need to be in the same helmfile. Example:
```yaml
  - name: thanos
    namespace: thanos
    needs: [maintenance/wait-for-minio]
    ...
  - name: wait-for-minio
    installed: {{ $a | get "minio.enabled" }}
    <<: *wait
    values:
      - url: {{ http://minio.minio }}
        host: minio.{{ $v.cluster | get "domainSuffix" nil }}
        tasksVersion: {{ $v.versions.tasks }}
        {{- if $v._derived.untrustedCA }}
        customCa: |
          {{- $v._derived.caCert | nindent 10 }}
        {{- end }}
        runPolicy: OnSpecChange    

```

File: helmfile.d/snippets/defaults.yaml [hunk 1]
```diff
@@ -143,24 +150,113 @@ environments:
...
+                  memory: 256Mi
+            compactor:
+              resources:
+                requests:
+                  cpu: 200m
+                  memory: 128Mi
+                limits:
+                  cpu: 500m
+                  memory: 256Mi
+              retentionResolutionRaw: 30d
```
Comment (j-zimnowoda) [inline helmfile.d/snippets/defaults.yaml:220]: > As a rule of thumb retention for each downsampling level should be the same, and should be greater than the maximum date range (10 days for 5m to 1h downsampling).
https://thanos.io/tip/components/compact.md/#-downsampling-note-about-resolution-and-retention-

Could you explain why do you chose `retentionResolution1h: 10y`

File: helmfile.d/snippets/defaults.yaml [hunk 2]
```diff
@@ -143,24 +150,113 @@ environments:
...
+              retentionResolution5m: 30d
+              retentionResolution1h: 10y
+              persistence:
+                size: 8Gi
```
Comment (j-zimnowoda) [inline helmfile.d/snippets/defaults.yaml:224]: > The compactor needs local disk space to store intermediate data for its processing as well as bucket state cache. Generally, for medium sized bucket about 100GB should be enough to keep working as the compacted time ranges grow over time.
https://thanos.io/tip/components/compact.md/#disk

8 GB does not seem to be a good default for compactor

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
So far few comments. Overall way too many changes in one PR.

File: values/minio/minio.gotmpl [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+{{- if $v._derived.untrustedCA }}
+extraVolumes:       
+  - name: root-ca-1
+    secret:
+      secretName: root-ca
+  - name: root-ca-2
+    secret:
+      secretName: root-ca
+extraVolumeMounts:
```
Comment (j-zimnowoda) [inline values/minio/minio.gotmpl:37]: Could you add a comment in the code explaining why the same cert needs to be mounted twice?

File: values/minio/minio.gotmpl [hunk 4]
```diff
@@ -0,0 +1,127 @@
+{{- $v := .Values }}
+{{- $vl:= $v.apps.minio }}
```
Comment (j-zimnowoda) [inline values/minio/minio.gotmpl:2]: I believe this variable should be called `$m`

**Comment by srodenhuis** [comment]
Will be split up into 4 separate PRs

**Pull request closed without merging**